analyzer_test.go

```diff
@@ -9,34 +9,22 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/golang/dep/test"
 )
 
 func TestDeriveManifestAndLock(t *testing.T) {
-	dir, err := ioutil.TempDir("", "dep")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer os.RemoveAll(dir)
+	h := test.NewHelper(t)
+	defer h.Cleanup()
 
-	contents := `{
-    "dependencies": {
-        "github.com/pkg/errors": {
-            "version": ">=0.8.0, <1.0.0"
-        },
-        "github.com/sdboyer/gps": {
-            "version": ">=0.12.0, <1.0.0"
-        }
-    }
-}
-`
-
-	if err := ioutil.WriteFile(filepath.Join(dir, ManifestName), []byte(contents), 0644); err != nil {
-		t.Fatal(err)
-	}
+	h.TempDir("dep")
+	golden := "analyzer/manifest.json"
+	contents := h.GetTestFileString(golden)
```

Comment:
If we are going with consistency with other go tools I think `contents` should be `want` and `b` (down below) should be `got` which IMO also simplifies reading of the ifs below. With that said, there is a lot of those cases in the code where this should probably be cleaned up, so it's fine if we do that in a different PR too.